### ISSUE TITLE
Detailed spec setup instructions

### ISSUE THREAD

**Issue by acreilly** [boomerdigital/solidus_elastic_product#2]
I attempted to setup the gem locally, trying to run the specs so I could see how certain parts of the code works, but never successfully got the code up and running. Kept getting errors.
Steps I took:
```
bundle
bundle exec rake test_app
```
Response:
```
solidus_elastic_product [master] :> bundle exec rake test_app
Expected string default value for '--rc'; got false (boolean)
rake aborted!
NoMethodError: undefined method `last_comment' for #<Rake::Application:0x007ff708bdbcb0>
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rubocop-0.37.2/lib/rubocop/rake_task.rb:24:in `initialize'
/Users/allisonreilly/Assets/gems/solidus_elastic_product/Rakefile:12:in `new'
/Users/allisonreilly/Assets/gems/solidus_elastic_product/Rakefile:12:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)
```

I upgraded rubocop to the latest version 1.47.1...
```
solidus_elastic_product [master] :> bundle exec rake test_app
Expected string default value for '--rc'; got false (boolean)
Generating dummy Rails application...
Setting up dummy database...
bundle exec rake db:migrate VERBOSE=false
Copying order bill addresses to credit cards
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

SQLite3::SQLException: cannot rollback - no transaction is active: rollback transaction
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `step'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `block in each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `loop'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `map'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `block in execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:95:in `prepare'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:137:in `execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:527:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `block in exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:266:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/query_cache.rb:14:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:145:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:196:in `rescue in within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:193:in `within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/transactions.rb:220:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:1044:in `ddl_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:959:in `block in migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:823:in `up'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:801:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
ActiveRecord::StatementInvalid: SQLite3::SQLException: cannot rollback - no transaction is active: rollback transaction
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `step'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `block in each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `loop'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `map'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `block in execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:95:in `prepare'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:137:in `execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:527:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `block in exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:266:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/query_cache.rb:14:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:145:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:196:in `rescue in within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:193:in `within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/transactions.rb:220:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:1044:in `ddl_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:959:in `block in migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:823:in `up'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:801:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
SQLite3::SQLException: cannot rollback - no transaction is active
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `step'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `block in each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `loop'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `map'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `block in execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:95:in `prepare'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:137:in `execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:527:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `block in exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:349:in `exec_rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:266:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/query_cache.rb:14:in `rollback_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:145:in `rollback'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:196:in `rescue in within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:193:in `within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/transactions.rb:220:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:1044:in `ddl_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:959:in `block in migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:823:in `up'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:801:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
ActiveRecord::StatementInvalid: SQLite3::SQLException: cannot commit - no transaction is active: commit transaction
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `step'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `block in each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `loop'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `map'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `block in execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:95:in `prepare'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:137:in `execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:518:in `commit'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:345:in `block in commit_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:345:in `commit_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:151:in `commit'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:175:in `commit_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:194:in `within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/transactions.rb:220:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:1044:in `ddl_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:959:in `block in migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:823:in `up'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:801:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
SQLite3::SQLException: cannot commit - no transaction is active
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `step'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:108:in `block in each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `loop'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/statement.rb:107:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `map'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:152:in `block in execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:95:in `prepare'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:137:in `execute'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:518:in `commit'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:345:in `block in commit_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/sqlite3_adapter.rb:345:in `commit_db_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:151:in `commit'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:175:in `commit_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/transaction.rb:194:in `within_new_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/transactions.rb:220:in `transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:1044:in `ddl_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:959:in `block in migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `each'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:955:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:823:in `up'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/migration.rb:801:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1@global/gems/activerecord-4.2.7.1/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [bundle exec rake db:migrate VERBOSE=false...]
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/solidus_core-1.4.0/lib/spree/testing_support/common_rake.rb:20:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/solidus_core-1.4.0/lib/spree/testing_support/extension_rake.rb:7:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/Assets/gems/solidus_elastic_product/Rakefile:29:in `block in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => common:test_app
(See full trace by running task with --trace)
```

I tried multiple ways of changing the solidus versions.. and rake test_app doesn't work but returns a different response

```
solidus_elastic_product [master] :> rake test_app
Generating dummy Rails application...
Setting up dummy database...
bin/rails db:environment:set RAILS_ENV=test
Error: Command 'db:environment:set' not recognized
Usage: rails COMMAND [ARGS]

The most common rails commands are:
 generate    Generate new code (short-cut alias: "g")
 console     Start the Rails console (short-cut alias: "c")
 server      Start the Rails server (short-cut alias: "s")
 dbconsole   Start a console for the database specified in config/database.yml
             (short-cut alias: "db")
 new         Create a new Rails application. "rails new my_app" creates a
             new application called MyApp in "./my_app"

In addition to those, there are:
 destroy      Undo code generated with "generate" (short-cut alias: "d")
 plugin new   Generates skeleton for developing a Rails plugin
 runner       Run a piece of code in the application environment (short-cut alias: "r")

All commands can be run with -h (or --help) for more information.
rake aborted!
Command failed with status (1): [bin/rails db:environment:set RAILS_ENV=tes...]
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/solidus_core-2.1.0/lib/spree/testing_support/common_rake.rb:20:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/solidus_core-2.1.0/lib/spree/testing_support/extension_rake.rb:7:in `block (2 levels) in <top (required)>'
/Users/allisonreilly/Assets/gems/solidus_elastic_product/Rakefile:29:in `block in <top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/Users/allisonreilly/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => common:test_app
(See full trace by running task with --trace)
```

I also tried google, but it didn't help.

**Comment by mtomov** [comment]
Hi Allie, thanks for taking the time to try to set it up. 

I think that both issues should now be resolved with f15fa68 and 6b8fdad. I've also updated the instructions to specify that `rake test_app` need be run to set-up the gem.

Let me know if all is good now.

**Comment by acreilly** [comment]
That fixed it, thanks!
